src/vm/runtimecallablewrapper.cpp

```diff
@@ -131,127 +129,109 @@ IUnknown *ComClassFactory::CreateInstanceFromClassFactory(IClassFactory *pClassF
     }
     else
     {
-        if (m_pClassMT == NULL)
+        _ASSERTE(m_pClassMT != NULL);
+
+        // Get the type to query for licensing.
+        TypeHandle rth = TypeHandle(m_pClassMT);
+
+        struct
         {
-            // Create an instance of the object.
-            FrameWithCookie<DebuggerExitFrame> __def;
+            OBJECTREF pProxy;
+            OBJECTREF pType;
+        } gc;
+        gc.pProxy = NULL; // LicenseInteropProxy
+        gc.pType = NULL;
+
+        GCPROTECT_BEGIN(gc);
+
+        // Create an instance of the object
+        MethodDescCallSite createObj(METHOD__LICENSE_INTEROP_PROXY__CREATE);
+        gc.pProxy = createObj.Call_RetOBJECTREF(NULL);
+        gc.pType = rth.GetManagedClassObject();
+
+        // Query the current licensing context
+        MethodDescCallSite getCurrentContextInfo(METHOD__LICENSE_INTEROP_PROXY__GETCURRENTCONTEXTINFO, &gc.pProxy);
+        BOOL fDesignTime = FALSE;
```

Comment:
```suggestion
        CLR_BOOL fDesignTime = FALSE;
```

Reply:
This seems wrong to me. Isn't the `CLR_BOOL` only for `FCall`s? What I am doing here seems more like a reverse P/Invoke so `QCall`, which should be `BOOL` right?

Reply:
Actually is this because the value is being set in the managed side which would be of a `CLR_BOOL`?

Reply:
Yes. There is no PInvoke marshaling here.